### PR TITLE
[KL-513] Prepare for shelter audit

### PR DESCRIPTION
**Pull request by thesvyatoslavtt**

### CODE REVIEW

File: libs/shared/src/index.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export {};
```
Comment (MuKoJla) [inline libs/shared/src/index.ts]: do you need this file?

File: libs/shelter/package.json [hunk 2]
```diff
@@ -0,0 +1,35 @@
+{
```
Comment (MuKoJla) [inline libs/shelter/package.json:1]: can not run app on mobile 
image

File: libs/shelter/package.json [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  "version": "1.0.0",
+  "main": "src/index.ts",
+  "license": "MIT",
+  "scripts": {
+    "ts": "tsc --pretty"
+  },
+  "dependencies": {
+    "react": "^18.2.0",
+    "react-redux": "^8.0.5",
+    "@reduxjs/toolkit": "^1.9.1",
```
Comment (MuKoJla) [inline libs/shelter/package.json]: do you need this libs?

File: libs/shelter/package.json [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    "redux-persist": "^6.0.0",
+    "react-native-themis": "^0.14.11",
+    "@react-native-async-storage/async-storage": "^1.17.11",
+    "@rnw-community/shared": "^0.48.0",
+    "@taquito/signer": "^13.0.1",
+    "bip39": "^3.0.4",
+    "ethers": "^5.7.2",
+    "rxjs": "^7.8.0",
+    "react-native-toast-notifications": "^3.3.1",
```
Comment (MuKoJla) [inline libs/shelter/package.json]: do you need this lib?

File: apps/extension/webpack.config.js [hunk 5]
```diff
@@ -74,12 +74,15 @@ module.exports = {
...
+            'react-native$': 'react-native-web',
+            'shared': path.resolve(__dirname, '../../libs/shared/src'),
+            'shelter': path.resolve(__dirname, '../../libs/shelter/src'),
         },
         plugins: [new TsconfigPathsPlugin()],
         modules: [
             path.resolve(__dirname, 'node_modules'),
-            path.resolve(__dirname, '../../libs/ui/node_modules')
+            path.resolve(__dirname, '../../libs/ui/node_modules'),
+            path.resolve(__dirname, '../../libs/shelter/node_modules'),
```
Comment (unixvb) [inline apps/extension/webpack.config.js]: create array of external libs ['ui', 'shelter', 'shelter'] and do map for their aliases and modules

File: apps/mobile/metro.config.js [hunk 6]
```diff
@@ -19,12 +19,15 @@ module.exports = (async () => {
...
+        path.resolve(__dirname, '../../libs/shared/src'),
+        path.resolve(__dirname, '../../libs/shelter/src'),
     ],
     resolver: {
         disableHierarchicalLookup: true,
         nodeModulesPaths: [
             path.resolve(__dirname, 'node_modules'),
-            path.resolve(__dirname, '../../libs/ui/node_modules')
+            path.resolve(__dirname, '../../libs/ui/node_modules'),
+            path.resolve(__dirname, '../../libs/shelter/node_modules'),
```
Comment (unixvb) [inline apps/mobile/metro.config.js]: the same here

File: libs/shared/src/utils/store.util.ts [hunk 7]
```diff
@@ -1,9 +1,9 @@
 import { isDefined } from '@rnw-community/shared';
 
-import { LocalStorage } from './local-storage.util';
+import { LocalStorage } from '../../../shelter/src/utils/local-storage.util';
```
Comment (unixvb) [inline libs/shared/src/utils/store.util.ts]: `shared` should not depend on `shelter`, only `shelter` on `shared`

File: libs/shelter/src/themis/decrypt.ts [hunk 8]
```diff
@@ -1,6 +1,6 @@
 import { SymmetricKey, initialized, SecureCellSeal } from 'wasm-themis';
 
-import { getStoredValue } from '../utils/store.util';
+import { getStoredValue } from '../../../shared/src/utils/store.util';
```
Comment (unixvb) [inline libs/shelter/src/themis/decrypt.ts]: use alias, as you do in other places

File: libs/shelter/src/utils/sign-message-by-method.util.ts [hunk 9]
```diff
@@ -2,7 +2,7 @@ import { signTypedData, SignTypedDataVersion } from '@metamask/eth-sig-util';
 import { ethers } from 'ethers';
 import { of } from 'rxjs';
 
-import { DAppMethodEnum } from '../../enums/dApp-method.enum';
+import { DAppMethodEnum } from '../../../ui/src/enums/dApp-method.enum';
```
Comment (unixvb) [inline libs/shelter/src/utils/sign-message-by-method.util.ts]: `shelter` could not depend on `ui`, only `ui` -> `shelter`  
and use aliases

File: libs/shelter/src/utils/tezos-toolkit.utils.ts [hunk 10]
```diff
@@ -1,18 +1,15 @@
...
 
 export const createTezosToolkit = (rpcUrl: string) => new TezosToolkit(rpcUrl);
 
 export const createReadOnlyTezosToolkit = (rpcUrl: string, sender: AccountInterface) => {
   const readOnlyTezosToolkit = createTezosToolkit(rpcUrl);
 
-  const publicKey = getString(sender.networksKeys[NetworkTypeEnum.Tezos]?.publicKey);
-  const publicKeyHash = getString(sender.networksKeys[NetworkTypeEnum.Tezos]?.publicKeyHash);
+  const publicKey = sender.networksKeys[NetworkTypeEnum.Tezos]?.publicKey ?? '';
+  const publicKeyHash = sender.networksKeys[NetworkTypeEnum.Tezos]?.publicKeyHash ?? '';
```
Comment (unixvb) [inline libs/shelter/src/utils/tezos-toolkit.utils.ts:12]: how is the case when there is no `networksKeys`  handled?

File: libs/ui/package.json [hunk 11]
```diff
@@ -41,10 +40,8 @@
     "react-native-device-info": "^10.3.0",
     "react-native-gesture-handler": "^2.8.0",
     "react-native-haptic-feedback": "^1.14.0",
-    "react-native-keychain": "^8.1.1",
     "react-native-qrcode-svg": "^6.1.2",
     "react-native-safe-area-context": "^4.4.1",
-    "react-native-sha256": "^1.4.9",
```
Comment (unixvb) [inline libs/ui/package.json:47]: are you sure you did remove all dependencies that are unused in `ui` anymore?

File: libs/ui/src/background-script.ts [hunk 12]
```diff
@@ -4,14 +4,11 @@
 //  3. get rid of ui/background-script imports
 //  4. extract this packages into separate libs
 
-export type { BackgroundMessage } from './messagers/types/background-message.type';
+export { BackgroundMessageType, E2eMessageType, INITIAL_PASSWORD_HASH, LocalStorage } from 'shelter';
+export type { BackgroundMessage } from 'shelter';
```
Comment (unixvb) [inline libs/ui/src/background-script.ts]: I think `shelter` lib should have file `background-script.ts)` with seperate exports for background script

File: libs/shelter/tsconfig.json [hunk 13]
```diff
@@ -0,0 +1,24 @@
+{
+  "compilerOptions": {
+    "baseUrl": "../",
```
Comment (unixvb) [inline libs/shelter/tsconfig.json]: why `baseUrl` not `./`?

File: libs/ui/tsconfig.json [hunk 14]
```diff
@@ -1,17 +1,12 @@
 {
   "compilerOptions": {
-    "baseUrl": "./",
+    "baseUrl": "../",
```
Comment (unixvb) [inline libs/ui/tsconfig.json]: why `baseUrl` is not `./`?

**Review by unixvb — DISMISSED** [review]

**Review by Boltman92 — DISMISSED** [review]

**Review by unixvb — APPROVED** [review]

**Review by vladvn1905 — APPROVED** [review]

**Pull request merged**